### PR TITLE
Implement AbstractPrivateMethodRule

### PR DESCRIPTION
**Pull request by staabm**
Abstract function cannot be declared private

https://3v4l.org/9V2aN#veol

----

as a followup I would do similar stuff for interfaces: https://3v4l.org/3VBl2

### CODE REVIEW

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
1) This has to be allowed in traits
2) Please test interfaces too

File: src/Rules/Methods/AbstractPrivateMethodRule.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+		if (!$classReflection->isAbstract() && !$classReflection->isInterface()) {
+			return [];
+		}
+
+		return [
+			RuleErrorBuilder::message(sprintf(
+				'Private method %s::%s() cannot be abstract.',
+				$method->getDeclaringClass()->getDisplayName(),
+				$method->getName(),
```
Comment (ondrejmirtes) [inline src/Rules/Methods/AbstractPrivateMethodRule.php:50]: Non-ignorable please

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Thank you. Please do final private next.

**Comment by staabm** [comment]
> Please do final private next.

what do you have in mind what https://github.com/phpstan/phpstan-src/blob/1.11.x/src/Rules/Methods/FinalPrivateMethodRule.php is not already doing?

**Comment by ondrejmirtes** [comment]
oh sorry, I didn't know we already have it

**Pull request merged**